_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+CONTEXT=$(kubectl config current-context)
+
+kubectl apply --context $CONTEXT \
+    -f ./spinnaker-service-account.yml
+
+
+TOKEN=$(kubectl get secret --context $CONTEXT \
+   $(kubectl get serviceaccount spinnaker-service-account \
+       --context $CONTEXT \
+       -n spinnaker \
+       -o jsonpath='{.secrets[0].name}') \
+   -n spinnaker \
+   -o jsonpath='{.data.token}' | base64 --decode)
+
+kubectl config set-credentials ${CONTEXT}-token-user --token $TOKEN
+
+kubectl config set-context $CONTEXT --user ${CONTEXT}-token-user
+```
+Enable the kubernetes provider in Halyard:
+
+```shell
+hal config provider kubernetes enable
+```
+
+Add the service account to the Halyard configuration:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+hal config provider kubernetes account add spinnaker-gke-account \
+    --context $CONTEXT
+```
+
+### Choose the installation environment
+
+The __Distributed installation__ on Kubernetes deploys each Spinnaker's microservice separately to a remote cloud, and it is the recommended environment for zero-downtime updates of Spinnaker.
+
+Select the distributed deployment with:
+
+```shell
+hal config deploy edit --type distributed --account-name spinnaker-gke-account
+```
+
+### Choose a storage service
+
+Spinnaker requires an external storage provider for persisting application settings and configured pipelines. To avoid losing this data, using a hosted storage solution is recommended. For this example, use Google Cloud Storage. Hosting the data externally allows to delete clusters in between sessions, and keep the pipelines once the clusters have been recreated.
+
+In the `spinnaker` folder, run the following lines:
+
+```shell
+SERVICE_ACCOUNT_NAME=spinnaker-gcs-account
+SERVICE_ACCOUNT_DEST=~/.gcp/gcs-account.json
+
+gcloud iam service-accounts create \
+    $SERVICE_ACCOUNT_NAME \
+    --display-name $SERVICE_ACCOUNT_NAME
+
+SA_EMAIL=$(gcloud iam service-accounts list \
+    --filter="displayName:$SERVICE_ACCOUNT_NAME" \
+    --format='value(email)')
+
+PROJECT=$(gcloud config get-value project)
+
+gcloud projects add-iam-policy-binding $PROJECT \
+    --role roles/storage.admin --member serviceAccount:$SA_EMAIL
+
+mkdir -p $(dirname $SERVICE_ACCOUNT_DEST)
+
+gcloud iam service-accounts keys create $SERVICE_ACCOUNT_DEST \
+    --iam-account $SA_EMAIL
+
+BUCKET_LOCATION=us
+
+hal config storage gcs edit --project $PROJECT \
+    --bucket-location $BUCKET_LOCATION \
+    --json-path $SERVICE_ACCOUNT_DEST
+
+hal config storage edit --type gcs    
+```
+
+The last `hal` commands edit the storage settings and set the storage source to GCS.
+
+### Deploy Spinnaker
+
+You can verify the deployment configuration with:
+
+```shell
+hal config deploy
+```
+
+And the storage configuration with:
+```shell
+hal config storage
+```
+
+List the available Spinnaker versions:
+
+```shell
+hal version list
+```
+
+Set the version for the deployment with:
+
+```shell
+hal config version edit --version $VERSION
+```
```

Comment:
I expect some advice here. What version did you test with? When I try this command, it fails:

```
Expected a value after parameter --version
```

Reply:
I tested with `hal config version edit --version 1.27.0`.

Reply:
Updated the version.